From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Mon, 28 Mar 2022 15:25:41 +0200
Subject: shellDBus: Actually make ScreenTransitionAsync async

We called such function async but we didn't awaited for the
checkInvocation result in the proper way.

So on failure the shell would report an unhandled promise rejection.

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2258
---
 js/ui/shellDBus.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ui/shellDBus.js b/js/ui/shellDBus.js
index 8faa112..edc16eb 100644
--- a/js/ui/shellDBus.js
+++ b/js/ui/shellDBus.js
@@ -253,9 +253,9 @@ var GnomeShell = class {
         invocation.return_value(GLib.Variant.new('(b)', [ungrabSucceeded]));
     }
 
-    ScreenTransitionAsync(params, invocation) {
+    async ScreenTransitionAsync(params, invocation) {
         try {
-            this._senderChecker.checkInvocation(invocation);
+            await this._senderChecker.checkInvocation(invocation);
         } catch (e) {
             invocation.return_gerror(e);
             return;
